File: sort-mapper-example-java/src/main/java/kh/farrukh/sortmapperexamplejava/common/exception/BaseException.java
package kh.farrukh.sortmapperexamplejava.common.exception;

import java.time.LocalDateTime;
import lombok.Getter;
import lombok.Setter;
import org.springframework.http.HttpStatus;

@Getter
@Setter
public abstract class BaseException extends RuntimeException {

  private String message;
  private HttpStatus httpStatus;
  private LocalDateTime time;

  public BaseException(String message, HttpStatus httpStatus, Throwable cause) {
    super(message, cause);
    this.message = message;
    this.httpStatus = httpStatus;
    this.time = LocalDateTime.now();
  }
}

File: sort-mapper-example/src/main/kotlin/kh/farrukh/sortmapperexample/common/exception/BaseException.kt
package kh.farrukh.sortmapperexample.common.exception

import org.springframework.http.HttpStatus
import java.time.LocalDateTime

abstract class BaseException(
    override val message: String,
    val httpStatus: HttpStatus,
    override val cause: Throwable? = null,
    val time: LocalDateTime = LocalDateTime.now()
) : RuntimeException(
    message, cause
)